File: consumerGo/internal/interfaces/message.go
package interfaces

import "time"

type MessageTemplateProps[T any] struct {
	Id      string    `json:"id"`
	Type    string    `json:"type"`
	Date    time.Time `json:"date"`
	Payload T         `json:"payload"`
}

type MessageInfoQuantityProps struct {
	UserId     string `json:"userId"`
	CompanyId  string `json:"companyId"`
	CampaignId string `json:"campaignId"`
	Quantity   int    `json:"quantity"`
}

type IMessage struct {
	UserId      string `json:"userId"`
	CompanyId   string `json:"companyId"`
	CampaignId  string `json:"campaignId"`
	PhoneNumber string `json:"phoneNumber"`
	Message     string `json:"message"`
}

File: consumerRust/src/interfaces/message.rs
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize, Debug)]
pub struct MessageTemplateProps {
    pub id: String,
    pub type_message: String,
    pub date: String,
    pub payload: serde_json::Value
}
#[derive(Serialize, Deserialize, Debug)]
pub struct  MessagePropsProps {
    pub message: IMessage,
    pub userId: String,
    pub companyId: String
}
#[derive(Serialize, Deserialize, Debug)]
pub struct  MessageInfoQuantityProps {
    pub quantity: i64,
    pub userId: String,
    pub companyId: String,
    pub campaignId: String,
}
#[derive(Serialize, Deserialize, Debug)]
pub struct  IMessage {
    pub id: String,
    pub phoneNumber: String,
    pub message: String,
    pub campaignId: String,
    pub userId: String,
    pub companyId: String,
    pub createdAt: String,
    pub updatedAt: String,
    pub deleted: bool,
}